samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -402,6 +402,39 @@ mfxStatus CDecodingPipeline::Init(sInputParams *pParams)
     sts = m_mfxSession.QueryIMPL(&m_impl); // get actual library implementation
     MSDK_CHECK_STATUS(sts, "m_mfxSession.QueryIMPL failed");
 
+    m_bVppIsUsed = IsVppRequired(pParams);
+    if (m_bVppIsUsed)
+        m_bDecOutSysmem = pParams->bUseHWLib ? false : true;
+    else
+        m_bDecOutSysmem = pParams->memType == SYSTEM_MEMORY;
+
+    m_eWorkMode = pParams->mode;
+
+    m_monitorType = pParams->monitorType;
+    // create device and allocator
+#if defined(LIBVA_SUPPORT)
+    m_libvaBackend = pParams->libvaBackend;
+#endif // defined(MFX_LIBVA_SUPPORT)
+
+#ifdef LIBVA_SUPPORT
+    if(m_memType != SYSTEM_MEMORY || !m_bDecOutSysmem
```

Comment:
For VAAPI path, we don't need this "if ()"

Reply:
Done